workflow/controller/exit_handler.go

```diff
@@ -24,43 +32,75 @@ func (woc *wfOperationCtx) runOnExitNode(ctx context.Context, exitHook *wfv1.Lif
 		execute := true
 		var err error
 		if exitHook.Expression != "" {
-			execute, err = argoexpr.EvalBool(exitHook.Expression, env.GetFuncMap(template.EnvMap(woc.globalParams.Merge(scope.getParameters()))))
+			scopeParams := scope.getParameters()
+			mergedParams := woc.globalParams.Merge(scopeParams)
```

Comment:
Does Scope not contains the global param? if yes, we should add it caller function. Scope should have all the paramers